frontend/src/app/appbar/appbar.component.ts

```diff
@@ -0,0 +1,25 @@
+import { Component, OnInit } from '@angular/core';
+import {TokenService} from "../authentication/service/token.service";
+
+@Component({
+  selector: 'app-appbar',
+  templateUrl: './appbar.component.html',
+  styleUrls: ['./appbar.component.sass']
+})
+export class AppbarComponent implements OnInit {
+
+  constructor(private authService: TokenService) { }
+
+  ngOnInit(): void {
+  }
+
+  changeLanguage(code: string) {
+      localStorage.setItem('locale', code);
+      window.location.reload();
```

Comment:
This seems like a workaround. However, for the sake of time savings I'd consider it "good enough". Just wanted to mention it in case there is a cleaner way to handle re-rendering.